docs/register_agreement.yml

```diff
@@ -0,0 +1,253 @@
+                    "0xE749754EC8CF03b1b9E10B81270B73AF66E40F54",
+                    "0xCCaF6665eb9Eb107c989DeA1Ad9Fd92DB940A504"
+                  ]
+                }
+              ],
+              "events": [
+                {
+                  "name": "Fulfilled",
+                  "actorType": "publisher",
+                  "handler": {
+                    "moduleName": "lockPaymentCondition",
+                    "functionName": "fulfillTransferNFTCondition",
+                    "version": "0.1"
+                  }
+                }
+              ]
+            },
+            {
+              "name": "transferNFT",
+              "timelock": 0,
+              "timeout": 0,
+              "contractName": "TransferNFTCondition",
+              "functionName": "fulfill",
+              "parameters": [
+                {
+                  "name": "_documentId",
+                  "type": "bytes32",
+                  "value": "249218ab779e6a16cd3ea1c630e7d97531343a1271cd5c16921e2491a65248c9"
+                },
+                {
+                  "name": "_receiver",
+                  "type": "address",
+                  "value": ""
+                },
+                {
+                  "name": "_numberNfts",
+                  "type": "uint256",
+                  "value": "1"
+                },
+                {
+                  "name": "_nftHolder",
+                  "type": "address",
+                  "value": "0xE749754EC8CF03b1b9E10B81270B73AF66E40F54"
+                },
+                {
+                  "name": "_conditionId",
+                  "type": "bytes32",
+                  "value": ""
+                }
+              ],
+              "events": [
+                {
+                  "name": "Fulfilled",
+                  "actorType": "publisher",
+                  "handler": {
+                    "moduleName": "transferNFT",
+                    "functionName": "fulfillEscrowPaymentCondition",
+                    "version": "0.1"
+                  }
+                },
+                {
+                  "name": "TimedOut",
+                  "actorType": "consumer",
+                  "handler": {
+                    "moduleName": "access",
+                    "functionName": "fulfillEscrowPaymentCondition",
+                    "version": "0.1"
+                  }
+                }
+              ]
+            },
+            {
+              "name": "escrowPayment",
+              "timelock": 0,
+              "timeout": 0,
+              "contractName": "EscrowPaymentCondition",
+              "functionName": "fulfill",
+              "parameters": [
+                {
+                  "name": "_did",
+                  "type": "bytes32",
+                  "value": "249218ab779e6a16cd3ea1c630e7d97531343a1271cd5c16921e2491a65248c9"
+                },
+                {
+                  "name": "_amounts",
+                  "type": "uint256[]",
```

Comment:
@aaitor If I understood it correctly then whenever there's a new agreement put in place the new price(new resell price) will always be integrated here in the parameters of the conditions, for each of the conditions in there. So this way we have an organic data source that's getting the data directly from the Agreement itself. And in the sdk we get the object itself and dump it in here.

Reply:
yes, the new agreement in the secondary market needs the price, but we don't need to record all the other DDO structure that is not required for initializing the service agreement. I think with a new model representing the 5 or 6 parameters necessary would be enough for having a simple index and not replicating the DDO structure that is too complex and is not necessary here.

Reply:
👍  I agree too

Reply:
Let me quickly reduce it

Reply:
@aaitor  now it's a lot more concise 👍

Reply:
looking good